rest/route/service_test.go

```diff
@@ -1162,6 +1163,63 @@ func TestTaskResetExecute(t *testing.T) {
 
 }
 
+func TestParentTaskInfo(t *testing.T) {
+	assert.NoError(t, db.ClearCollections(task.Collection))
+
+	sc := data.MockConnector{
+		URL: "http://evergreen.example.net",
+	}
+	buildID := "test"
+	dtID := "displayTask"
+	displayTask := task.Task{
+		Id:             dtID,
+		BuildId:        buildID,
+		DisplayOnly:    true,
+		ExecutionTasks: []string{"execTask0", "execTask1"},
+	}
+	execTask0 := task.Task{
+		Id:      "execTask0",
+		BuildId: buildID,
+	}
+	execTask1 := task.Task{
+		Id:      "execTask1",
+		BuildId: buildID,
+	}
+	randomTask := task.Task{
+		Id:      "randomTask",
+		BuildId: buildID,
+	}
+
+	assert.NoError(t, displayTask.Insert())
+	sc.MockTaskConnector.CachedTasks = append(sc.MockTaskConnector.CachedTasks, displayTask, execTask0, execTask1, randomTask)
+	ctx := context.Background()
+	tbh := &tasksByBuildHandler{
+		limit: 100,
+		sc:    &sc,
+	}
+	route := "/rest/v2/builds/test/tasks?fetch_all_executions=false&fetch_parent_ids=true&start_at=execTask0"
+	r, err := http.NewRequest("GET", route, nil)
+	assert.NoError(t, err)
+	r = gimlet.SetURLVars(r, map[string]string{"build_id": "test"})
+
+	tbh.Parse(ctx, r)
+	assert.False(t, tbh.fetchAllExecutions)
+	assert.True(t, tbh.fetchParentIds)
+
+	resp := tbh.Run(ctx)
+	data, ok := resp.Data().([]interface{})
+	assert.True(t, ok)
+	assert.Equal(t, 3, len(data))
+
+	expectedParentIDs := []string{dtID, dtID, ""}
+	for i := range data {
+		task, ok := data[i].(*model.APITask)
+		if assert.True(t, ok) {
```

Comment:
Actually, this should be true always, right? Making this an if statement means we maybe aren't even hitting 1218. I would think all three tasks are being returned and so they pass 1217, but 1218 should only be true for two of the tasks. I'm not sure this test should pass.